tiledb/sm/rest/curl.cc

```diff
@@ -180,21 +180,64 @@ static int buffer_list_seek_callback(
   return CURL_SEEKFUNC_FAIL;
 }
 
+/**
+ * This callback function gets called by libcurl as soon as a header has been
+ * received. libcurl buffers headers and delivers only "full" headers, one by
+ * one, to this callback. This callback should return the number of bytes
+ * actually taken care of if that number differs from the number passed to
+ * your callback function, it signals an error condition to the library
+ *
+ * @param res_data points to the delivered data
+ * @param size the size of that data is size multiplied with nmemb
+ * @param count number of bytes actually taken care of
+ * @param userdata the userdata argument is set with CURLOPT_HEADERDATA
+ * @return
+ */
+size_t OnReceiveData(
+    void* res_data, size_t size, size_t count, void* userdata) {
+  //  size_t length = size * count;
+  size_t index = size * count;
+  auto* buffer = static_cast<char*>(res_data);
+
+  //  while (index < length) {
+  //    char* temp = buffer + index;
+  //    if ((temp[0] == '\r') || (temp[0] == '\n'))
+  //      break;
+  //    index++;
+  //  }
+
+  std::string str(buffer, index);
+  size_t pos = str.find(": ");
+  auto* pmHeader = static_cast<HeaderCbData*>(userdata);
+  if (pos != std::string::npos) {
+    if (str.substr(0, pos) == constants::redirection_header_key) {
+      std::unique_lock<std::mutex> rd_lck(*(pmHeader->redirect_uri_map_lock));
+      pmHeader->redirect_uri_map->insert(
+          std::make_pair(pmHeader->uri, str.substr(pos + 2, index - 1)));
```

Comment:
This is keying by the full array URI, e.g. `https://multiregion-router-production.tiledb.workers.dev/v1/arrays/jmaley/test_arr`. We only want to key by the base URI: `https://multiregion-router-production.tiledb.workers.dev`.

Reply:
Wait, this is correct. Ignore this.